src/hypertable.c

```diff
@@ -118,18 +118,18 @@ ts_hypertable_permissions_check_by_id(int32 hypertable_id)
 }
 
 static Oid
-get_chunk_sizing_func_oid(FormData_hypertable *fd)
+get_chunk_sizing_func_oid(const FormData_hypertable *fd)
 {
 	Oid argtype[] = { INT4OID, INT8OID, INT8OID };
-	return LookupFuncName(list_make2(makeString(NameStr(fd->chunk_sizing_func_schema)),
-									 makeString(NameStr(fd->chunk_sizing_func_name))),
+	return LookupFuncName(list_make2(makeString(pstrdup(NameStr(fd->chunk_sizing_func_schema))),
+									 makeString(pstrdup(NameStr(fd->chunk_sizing_func_name)))),
```

Comment:
`makeString` expects a mutable string, so there is no guarantee that it won't change the string contents. Since the hypertable metadata is now `const` we need to make a copy of the string so that we know the original string is not modified by other code.

Reply:
Function `makeString` takes a mutable string because it is not `const`-correct, not because it intend to modify the string (it could just as well accept a `const char*`). The lifetime of the function name list is just for the duration of the lookup, so the copying is not necessary, especially not since this is intended to be a refactoring.

Reply:
Again, this PR does not address any lifetime issues. 

We can't on a case-by-case basis decide if a function is "const-correct", then we cannot rely on the interfaces given. We'd have to assume for safety that is is not.

You are also missing that the object returned by `makeString` could itself be passed to a function or be part of a containing object. So, even if `makeString` doesn't itself modify the string object within, another part of the code could. So, it is a false assumption to say that just because a wrapper around one object is not itself mutating the object within the wrapper then no other code can modify it.